evm/contracts/token-bridge/TokenBridge.sol

```diff
@@ -119,12 +136,24 @@ contract TokenBridge is LayerTransition{
         }
     }
 
+    /// @notice returns the withdraw limit
+    /// @return amount of tokens that can be withdrawn
+    function withdrawLimit() external view returns (uint256) {
+        if (block.timestamp - withdrawLimitUpdateTime > DEPOSIT_LIMIT_UPDATE_INTERVAL) {
+            return token.balanceOf(address(this)) / WITHDRAW_LIMIT_DENOMINATOR;
+        }
+        else{
+            return withdrawLimitRecord;
+        }
+    }
+
     /* Internal Functions */
     /// @notice refreshes the deposit limit every 12 hours so no one can spam layer with new tokens
-    function _refreshDepositLimit() internal returns (uint256) {
+    /// @return max amount of tokens that can be deposited
+    function _refreshDepositLimit(uint256 _amount) internal returns (uint256) {
         if (block.timestamp - depositLimitUpdateTime > DEPOSIT_LIMIT_UPDATE_INTERVAL) {
             uint256 _tokenBalance = token.balanceOf(address(this));
-            if (_tokenBalance < 100 ether) {
+            if (_tokenBalance < _amount) {
                 token.mintToOracle();
```

Comment:
is there a scenario where _tokenBalance could still be less than the amount?  Just checking here

Reply:
If mintToOracle hasn't been called in a while, and people have done a lot of withdrawals bringing the tokenBridge balance down a lot, tokenBalance could be lower than the desired deposit amount. 

tokenBalance==100
pending mintToOracle==1000
desired deposit==100

An alternative could be to use 
`if (_tokenBalance < _amount * 5)` for deposits

and 

`if (_tokenBalance < _amount * 20)` for withdraws